arch/arc/core/arc_smp.c

```diff
@@ -131,6 +131,19 @@ static int arc_smp_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	if (bcr.dbg) {
+	/* configure inter-core debug unit if available */
+		uint32_t core_mask = 0;
+		for (int i = 0; i < CONFIG_MP_NUM_CPUS; i++) {
+			core_mask |= 1 << i;
+		}
+		z_arc_connect_debug_select_set(core_mask);
+		/* Debugger halt cores at conditions */
+		z_arc_connect_debug_mask_set(core_mask,	(ARC_CONNECT_CMD_DEBUG_MASK_BH
+			| ARC_CONNECT_CMD_DEBUG_MASK_AH | ARC_CONNECT_CMD_DEBUG_MASK_H));
```

Comment:
I think we can choose all four conditions here as we don't have situations when core halts itself (except fatal error)